### PR TITLE
[test-utils] Support async onInitialized in TestRenderer

### PR DESCRIPTION
**Pull request by Pessimistress**
Align with `AnimationLoop`'s behavior.

#### Change List
- Allow `testCase.onInitialized` to return a promise
- Add more examples to render test

### CODE REVIEW

File: modules/test-utils/src/test-renderer.js [hunk 1]
```diff
@@ -158,17 +161,21 @@ export default class TestRenderer {
       this.currentTestCase = testCase;
       this._currentTestCaseStartTime = animationProps.time;
       this._currentTestCaseStartTick = animationProps.tick;
+      this._testCaseData = null;
       // initialize test case
       withParameters(animationProps.gl, DEFAULT_RENDER_PARAMETERS, () => {
-        this._testCaseData =
+        Promise.resolve(
```
Comment (georgios-uber) [inline modules/test-utils/src/test-renderer.js]: This resolves **immediately**
You can just do `this._testCaseData = ...`

Comment (Pessimistress) [reply modules/test-utils/src/test-renderer.js]: Not if `onInitialized` returns a promise.

Comment (georgios-uber) [reply modules/test-utils/src/test-renderer.js]: `(testCase.onInitialize() || Promise.resolve({})).then(...)`

Comment (georgios-uber) [reply modules/test-utils/src/test-renderer.js]: I think that way is easier to read.

Comment (Pessimistress) [reply modules/test-utils/src/test-renderer.js]: That won't work if `onInitialized` returns a plain object.

Comment (georgios-uber) [reply modules/test-utils/src/test-renderer.js]: OK, but can you add a comment?

**Pull request merged**